src/tests.rs

```diff
@@ -0,0 +1,284 @@
+use crate::benchmark::Benchmark;
+use crate::blending::{
+    blend_images, get_blending_algorithm, is_algorithm_multiplied, multiply_image, BlendAlgorithm,
+};
+use crate::errors::PConvertError;
+use crate::parallelism::{ResultMessage, ThreadPool};
+use crate::utils::{read_png_from_file, write_png_parallel, write_png_to_file};
+use image::codecs::png::{CompressionType, FilterType};
+use image::Rgba;
+use std::fmt;
+use std::fmt::{Display, Formatter};
+
+#[cfg(test)]
+use crate::constants;
+#[cfg(test)]
+use image::ImageFormat;
+#[cfg(test)]
+use std::str::FromStr;
+
+#[test]
+fn test_compose() {
+    let test_dir = "assets/test/";
+    let mut benchmark = Benchmark::new();
+    let backgrounds = vec![
+        Background::Alpha,
+        Background::Blue,
+        Background::Texture,
+        Background::White,
+    ];
+
+    // composes with different combinations of blending algorithms and backgrounds
+    for background in backgrounds {
+        for algorithm in constants::ALGORITHMS.iter() {
+            compose(
+                &test_dir,
+                BlendAlgorithm::from_str(algorithm).unwrap(),
+                background.clone(),
+                CompressionType::Fast,
+                FilterType::NoFilter,
+                &mut benchmark,
```

Comment:
Line 37 clones the background (rust specific, otherwise it would be moved out of the loop for first iteration)

Line 38 and 39 select the type of Compression and Filter to use. Because the combinatorial number of ways we can have the images with the backgrounds with the compression and filter types is to high, and because in production we use only Fast compression and NoFilter, I have left these hardcoded. But the function originally tested everything, so I left it generic.

Line 40 receives the benchmark struct to accumulate times.